vcs.go

```diff
@@ -0,0 +1,118 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+func repoDirExists(projPath, repoDir string) bool {
+	path := filepath.Join(projPath, repoDir)
+	info, err := os.Stat(path)
+	if err != nil {
+		return false
+	}
+	return info.IsDir()
+}
+
+// getId gets first line of commands output which should hold some VCS
+// specific id of current code checkout.
+func getId(dir, cmd string, params []string) (string, error) {
+	args := []string{cmd}
+	args = append(args, params...)
+	buffer := new(bytes.Buffer)
+	cmdPath, err := exec.LookPath(cmd)
+	if err != nil {
+		return "", nil
+	}
+	process := &exec.Cmd{
+		Path: cmdPath,
+		Args: args,
+		Env: []string{
+			"PATH=" + os.Getenv("PATH"),
+		},
+		Dir:    dir,
+		Stdout: buffer,
+	}
+	if err := process.Run(); err != nil {
+		return "", err
+	}
+	output := string(buffer.Bytes())
+	if newline := strings.Index(output, "\n"); newline < 0 {
+		return output, nil
+	} else {
+		return output[:newline], nil
+	}
+}
+
+func getLabelAndId(label, path, cmd string, params []string) (string, string, error) {
+	if info, err := getId(path, cmd, params); err != nil {
+		return "", "", err
+	} else {
+		return label, info, nil
+	}
+}
+
+type VCSInfo interface {
+	IsValid(path string) bool
+	GetLabelAndId(path string) (string, string, error)
```

Comment:
Awesome tests. Sad that the [vcs package](https://godoc.org/golang.org/x/tools/go/vcs) does not provide what we need here.

The sources of the vcs package are on https://github.com/golang/tools/tree/master/go/vcs but the issues are tracked on the [golang/go tracker](https://github.com/golang/go/issues).

Can you open an issue on the vcs package requesting the missing features and add a comment here in goaci with a link to the issue?